From f0c51af263e20f332c6f675aa90ec6705ae4f5d1 Mon Sep 17 00:00:00 2001
From: Serhii Abarovskyi <aserhii@protonmail.com>
Date: Tue, 9 May 2023 18:33:54 +0300
Subject: [PATCH] Use setuptools-scm v7+ for building the dists

Since version 7, setuptools-scm has native support for git archive,
so the setuptools-scm-git-archive project is obsolete
and this patch removes it from the build dependencies.
Close #515

Upstream: https://github.com/cherrypy/cheroot/commit/f0c51af263e20f332c6f675aa90ec6705ae4f5d1
Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
---
 .git_archival.txt                       |  3 +++
 pyproject.toml                          |  3 +--
 requirements/dist-build-constraints.in  |  3 +--
 requirements/dist-build-constraints.txt | 16 +++++++---------
 setup.cfg                               |  3 +--
 5 files changed, 13 insertions(+), 15 deletions(-)

diff --git a/.git_archival.txt b/.git_archival.txt
index 95cb3eea4e..3994ec0a83 100644
--- a/.git_archival.txt
+++ b/.git_archival.txt
@@ -1 +1,4 @@
+node: $Format:%H$
+node-date: $Format:%cI$
+describe-name: $Format:%(describe:tags=true)$
 ref-names: $Format:%D$
diff --git a/pyproject.toml b/pyproject.toml
index b1e5f60a80..88df57dcc1 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -6,8 +6,7 @@ requires = [
     "setuptools>=34.4",
 
     # Plugins
-    "setuptools_scm[toml]>=3.5",
-    "setuptools_scm_git_archive>=1.1",
+    "setuptools-scm >= 7.0.0",
 ]
 build-backend = "setuptools.build_meta"
 
diff --git a/requirements/dist-build-constraints.in b/requirements/dist-build-constraints.in
index 26b57228a9..a37baef1a9 100644
--- a/requirements/dist-build-constraints.in
+++ b/requirements/dist-build-constraints.in
@@ -5,8 +5,7 @@
 setuptools >= 34.4
 
 # Plugins
-setuptools_scm[toml] >= 3.5
-setuptools_scm_git_archive >= 1.1
+setuptools-scm[toml] >= 7.0.0
 
 # Dynamic (coming from setuptools' PEP 517 build backend)
 wheel
diff --git a/requirements/dist-build-constraints.txt b/requirements/dist-build-constraints.txt
index 525f2be30f..8bd4343ff6 100644
--- a/requirements/dist-build-constraints.txt
+++ b/requirements/dist-build-constraints.txt
@@ -1,18 +1,16 @@
 #
-# This file is autogenerated by pip-compile with python 3.10
-# To update, run:
+# This file is autogenerated by pip-compile with Python 3.10
+# by the following command:
 #
 #    pip-compile --allow-unsafe --output-file=requirements/dist-build-constraints.txt --strip-extras requirements/dist-build-constraints.in
 #
-packaging==21.3
+packaging==23.1
     # via setuptools-scm
-pyparsing==3.0.6
-    # via packaging
-setuptools-scm==6.3.2
+setuptools-scm==7.1.0
     # via -r requirements/dist-build-constraints.in
-setuptools-scm-git-archive==1.1
-    # via -r requirements/dist-build-constraints.in
-tomli==2.0.0
+tomli==2.0.1
+    # via setuptools-scm
+typing-extensions==4.6.3
     # via setuptools-scm
 wheel==0.37.1
     # via -r requirements/dist-build-constraints.in
diff --git a/setup.cfg b/setup.cfg
index 1f2b08f4c3..900c7b4feb 100644
--- a/setup.cfg
+++ b/setup.cfg
@@ -64,8 +64,7 @@ include_package_data = True
 packages = find:
 include_package_data = True
 setup_requires = 
-	setuptools_scm>=1.15.0
-	setuptools_scm_git_archive>=1.0
+	setuptools_scm >= 7.0.0
 install_requires = 
 	importlib_metadata; python_version < '3.8'
 	more_itertools >= 2.6
